src/hip_hcc.cpp

```diff
@@ -1619,7 +1619,9 @@ void ihipPrintKernelLaunch(const char* kernelName, const grid_launch_parm* lp,
 // Allows runtime to track some information about the stream.
 hipStream_t ihipPreLaunchKernel(hipStream_t stream, dim3 grid, dim3 block, grid_launch_parm* lp,
                                 const char* kernelNameStr, bool lockAcquired) {
-    stream = ihipSyncAndResolveStream(stream, lockAcquired);
+    if (stream == nullptr || stream != stream->getCtx()->_defaultStream){
+        stream = ihipSyncAndResolveStream(stream, lockAcquired);
```

Comment:
Could you clarify what issue these additional checks try to address?

Reply:
@scchan The ihipSyncAndResolveStream syncs a normal stream with a defaultStream and there are two cases:
1 - if a nullStream is passed, this function syncs with default stream and returns the defaultStream
2 - if a normal stream is passed, this function syncs the stream with defautlStream and returns the same normal stream.

The new additional checks make sure that only under the above two cases the ihipSyncAndResolveStream is called inside the ihipPreLaunchKernel. So, for example, if the defaultStream is passed, then it is not needed to call ihipSyncAndResolveStream here. 

This is the case where a nullStream is passed to hipLaunchCooperativeKernel API. In this case, there is already an ihipSyncAndResolveStream call inside the hipLaunchCooperativeKernel API to get the deafultStream so when we reached here in ihipPreLaunchKernel, it is not required to call ihipSyncAndResolveStream because we already have deafultStream and it is not needed to sync again with deafultStream itself.